charts/customer-center/templates/deployment-backend.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "customerCenter.fullname" . }}-backend
+  labels:
+    {{- include "customerCenter.labels" . | nindent 4 }}
+    app.kubernetes.io/component: backend
+spec:
+  replicas: {{ .Values.backend.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "customerCenter.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "customerCenter.labels" . | nindent 8 }}
+        app.kubernetes.io/component: backend
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}-backend
+          image: "{{ .Values.backend.image.repository }}:{{ default .Chart.AppVersion .Values.backend.image.tag }}"
+          envFrom:
+            - secretRef:
+                name: {{ include "customerCenter.fullname" . }}-backend
+            - configMapRef:
+                name: {{ include "customerCenter.fullname" . }}-backend
+          imagePullPolicy: {{ .Values.backend.image.pullPolicy }}
+          ports:
+            - containerPort: {{ .Values.backend.service.internalPort }}
+          {{- if .Values.backend.livenessProbe.enabled }}
+          livenessProbe:
+            initialDelaySeconds: {{ .Values.backend.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.backend.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.backend.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.backend.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.backend.livenessProbe.failureThreshold }}
+            httpGet:
+              path: {{ .Values.backend.livenessProbe.path }}
+              port: {{ .Values.backend.service.internalPort }}
+              httpHeaders:
+                - name: Host
+                  value: {{ mustFirst .Values.ingress.hosts | default "localhost" }}
```

Comment:
As discussed, this does not make any sense. We should stick with localhost. We don't want the liveness or readinessProbe to rely on the ingress functioning properly. It should be kept inside the pod. This is not an end2end check.

Reply:
Discard this, it's just the Header! setting. I'm dumb.

Reply:
Try it out on our AKS, If we don't need to set this headers then we shouldn't.

Reply:
After discussion and tests with @eyenx we decided to remove the useless headers from the backend deployment.